### PR TITLE
refactor(site): Highlight immutable parameters and do a few tweaks

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Video about the proposed changes:

https://user-images.githubusercontent.com/3165839/223513894-15029e51-1cd6-427b-b4f5-5468316fa29b.mov

But I decided to move forward with the banner option instead because we have two scenarios:
- Show the caption and primary text when the field has a name and description
- Only show the primary text when the field only has a name
Aligning the pill with these scenarios and making them look good was kinda hard. I also moved the icon to be left-center aligned with the text to make it easier to align with more generic cases.

Preview:
Screen Shot 2023-03-07 at 15 11 41

I added extra stories to make sure we are testing all the cases.

### CODE REVIEW

File: site/src/components/RichParameterInput/RichParameterInput.tsx [hunk 1]
```diff
@@ -70,16 +65,16 @@ export const RichParameterInput: FC<RichParameterInputProps> = ({
   onChange,
   parameter,
   initialValue,
-  ...props
+  ...fieldProps
```
Comment (Kira-Pilot) [inline site/src/components/RichParameterInput/RichParameterInput.tsx:68]: Why spread this? Isn't it just `id`?

File: site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx [hunk 2]
```diff
@@ -203,313 +206,200 @@ export const CreateWorkspacePageView: FC<
...
+        {/* Template params */}
+        {props.templateSchema && props.templateSchema.length > 0 && (
+          <FormSection
+            title="Template params"
+            description="Those values are provided by your template's Terraform
+            configuration."
+          >
+            <FormFields>
+              {props.templateSchema
+                // We only want to show schema that have redisplay_value equals true
```
Comment (Kira-Pilot) [inline site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx:279]: I like the addition of a comment here, but I'm still unclear as to why we don't want to show schemas with `redisplay_value=false`

File: site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx [hunk 3]
```diff
@@ -11,11 +10,17 @@ import { useTranslation } from "react-i18next"
...
 import { SelectedTemplate } from "./SelectedTemplate"
 import { Loader } from "components/Loader/Loader"
 import { GitAuth } from "components/GitAuth/GitAuth"
+import {
+  FormFields,
+  FormSection,
+  FormFooter,
+  HorizontalForm,
+} from "components/HorizontalForm/HorizontalForm"
+import { makeStyles } from "@material-ui/core/styles"
```
Comment (Kira-Pilot) [inline site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx:23]: I think it's probably time for this file to be broken apart.

**Review by Kira-Pilot — APPROVED** [review]
Nice! Thanks for the vid

File: site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx [hunk 4]
```diff
@@ -203,313 +206,200 @@ export const CreateWorkspacePageView: FC<
...
+              variant="outlined"
+            />
+          </FormFields>
+        </FormSection>
+
+        {/* Workspace owner */}
+        {props.canCreateForUser && (
+          <FormSection
+            title="Workspace owner"
+            description=" The user that is going to own this workspace. If you are
```
Comment (mtojek) [inline site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx:236]: nit: " The user <- starting spacebar

**Review by mtojek — APPROVED** [review]
Awesome!

File: site/src/components/RichParameterInput/RichParameterInput.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/components/RichParameterInput/RichParameterInput.tsx:68]: Nops, it passes other props. I probably should fix the types to make it clear... Going to make that.

File: site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx (hunk 3 above)
Comment (BrunoQuaresma) [reply site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx:23]: I think it is! I will do that.

**Comment by BrunoQuaresma** [comment]
Related to #6076

**Pull request merged**